homework-submission/myapp/src/Components/Data.js

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import Todos from './Todos.json';
+import Form from './Form';
+
+
+class Data extends React.Component {
+  state = {
+    Todos
+  };
+
+  descriptionInput = React.createRef();
+  deadlineInput = React.createRef();
+
+  addTodo = event => {
+    event.preventDefault();
+
+    const descriptionInput = this.descriptionInput.current.value;
+    const deadlineInput = this.deadlineInput.current.value;
+    const newState = this.state;
+
+    const Todos = newState.Todos;
+    Todos.push({
+      id: Todos.length + 1,
+      description: descriptionInput,
+      deadline: deadlineInput,
+      done: false
+    });
+
+    this.setState(newState);
+
+    this.descriptionInput.current.value = '';
+    this.deadlineInput.current.value = '';
+  };
+
+  deleteTodo = id => {
+    const arr = this.state.Todos;
+    arr.splice(id, 1);
+    this.setState({ Todos: arr });
+  };
+
+  isTaskDone = todo => {
+    todo.done = !todo.done;
+    this.setState({
+      todos: this.state.todos
+    });
+  };
+
+  renderTodos = () =>{
+    if (this.state.Todos.length === 0)
+      return <p className='noItems'>No items...</p>;
+    const textStyle = {
+      textDecoration: 'line-through',
+      backgroundColor: '#0455d8'
+    };
+
+    return (
+        <ul>
+        {Todos.map((todo, id) => {
```

Comment:
where `Todos` come from? 🙃

Reply:
Instead of `Todos.map` , my corrections are as follows `this.state.Todos.map` ... i hope im on the right track